central/node/globaldatastore/datastore_sac_test.go

```diff
@@ -0,0 +1,471 @@
+					sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+					sac.ResourceScopeKeys(resourceHandles...),
+					sac.ClusterScopeKeys(validClusterID),
+					sac.NamespaceScopeKeys("someNamespace"))),
+			SingleCluster:      true,
+			ExpectedClusterIds: []string{},
+		},
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetAllClusterNodeStores() {
+	clusterID := testconsts.Cluster2
+
+	cases := getReadSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+
+			results, err := s.globalDatastore.GetAllClusterNodeStores(ctx, false)
+			s.NoError(err)
+
+			fetchedClusterIDs := make([]string, 0, len(results))
+			for fetchedClusterID := range results {
+				fetchedClusterIDs = append(fetchedClusterIDs, fetchedClusterID)
+			}
+			s.ElementsMatch(c.ExpectedClusterIds, fetchedClusterIDs)
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetAllClusterNodeStoresWriteAccess() {
+	clusterID := testconsts.Cluster2
+
+	cases := getWriteSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+
+			results, err := s.globalDatastore.GetAllClusterNodeStores(ctx, true)
+			s.NoError(err)
+
+			fetchedClusterIDs := make([]string, 0, len(results))
+			for fetchedClusterID := range results {
+				fetchedClusterIDs = append(fetchedClusterIDs, fetchedClusterID)
+			}
+			s.ElementsMatch(c.ExpectedClusterIds, fetchedClusterIDs)
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetClusterNodeStore() {
+	clusterID := testconsts.Cluster2
+
+	cases := testutils.GenericClusterSACGetTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, false)
+			if c.ExpectedFound {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			} else {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetClusterNodeStoreWriteAccess() {
+	clusterID := testconsts.Cluster2
+
+	cases := testutils.GenericClusterSACWriteTestCases(context.Background(), s.T(), "write", clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, true)
+			if c.ExpectError {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			} else {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestRemoveClusterNodeStores() {
+	cases := testutils.GenericGlobalSACDeleteTestCases(s.T())
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := s.testContexts[c.ScopeKey]
+
```

Comment:
Technically, the tested function removes all nodes that are in the cluster whose IDs are passed as input.
IMHO, it would be good to test using a dedicated cluster ID for each test, inject nodes linked to that cluster, and check that the function does remove the nodes when it has the right to (and that it only removes the nodes it should remove).